dask/base.py

```diff
@@ -1605,4 +1608,51 @@ def clone_key(key, seed):
     raise TypeError(f"Expected str or tuple[str, Hashable, ...]; got {key}")
 
 
+def print_on_compute(msg: str, val: T) -> T:
+    """When the value is computed, print it using distributed.print.
+
+    Note: this may affect performance, since it will change the graph being
+    computed.
```

Comment:
It doesn't. It just adds an extra layer which will be promptly fused by the optimizer in most cases.

A much more important note is that this function will quietly collapse multi-chunk collections into a single chunk (as per base behaviour of dask.delayed). At the very least you should warn about this.

Reply:
> A much more important note is that this function will quietly collapse multi-chunk collections into a single chunk (as per base behaviour of dask.delayed). At the very least you should warn about this.

Is there a way to avoid it?

Reply:
No, because there's no collection-agnostic `map` function.